E0459 needs to be updated to new format

From: src/test/compile-fail/E0459.rs

E0459 needs a span_label, updating it from:

```
error[E0459]: #[link(...)] specified without `name = "foo"`
  --> src/test/compile-fail/E0459.rs:11:1
   |
11 | #[link(kind = "dylib")] extern {} //~ ERROR E0459
   | ^^^^^^^^^^^^^^^^^^^^^^^
```

To:

```
error[E0459]: #[link(...)] specified without `name = "foo"`
  --> src/test/compile-fail/E0459.rs:11:1
   |
11 | #[link(kind = "dylib")] extern {} //~ ERROR E0459
   | ^^^^^^^^^^^^^^^^^^^^^^^ missing `name` argument
```

--- a/src/librustc_metadata/creader.rs
+++ b/src/librustc_metadata/creader.rs
@@ -977,8 +977,9 @@ impl<'a> LocalCrateReader<'a> {
             let n = match n {
                 Some(n) => n,
                 None => {
-                    span_err!(self.sess, m.span, E0459,
-                              "#[link(...)] specified without `name = \"foo\"`");
+                    struct_span_err!(self.sess, m.span, E0459,
+                                     "#[link(...)] specified without `name = \"foo\"`")
+                        .span_label(m.span, &format!("missing `name` argument")).emit();
                     InternedString::new("foo")
                 }
             };

--- a/src/librustc_metadata/creader.rs
+++ b/src/librustc_metadata/creader.rs
@@ -965,8 +965,9 @@ impl<'a> LocalCrateReader<'a> {
                 Some("dylib") => cstore::NativeUnknown,
                 Some("framework") => cstore::NativeFramework,
                 Some(k) => {
-                    span_err!(self.sess, m.span, E0458,
-                              "unknown kind: `{}`", k);
+                    struct_span_err!(self.sess, m.span, E0458,
+                              "unknown kind: `{}`", k)
+                        .span_label(m.span, &format!("unknown kind")).emit();
                     cstore::NativeUnknown
                 }
                 None => cstore::NativeUnknown

--- a/src/test/compile-fail/E0458.rs
+++ b/src/test/compile-fail/E0458.rs
@@ -9,7 +9,9 @@
 // except according to those terms.
 
 #[link(kind = "wonderful_unicorn")] extern {} //~ ERROR E0458
-                                              //~^ ERROR E0459
+                                              //~| NOTE unknown kind
+                                              //~| ERROR E0459
+                                              //~| NOTE missing `name` argument
 
 fn main() {
 }

--- a/src/test/compile-fail/E0459.rs
+++ b/src/test/compile-fail/E0459.rs
@@ -9,6 +9,7 @@
 // except according to those terms.
 
 #[link(kind = "dylib")] extern {} //~ ERROR E0459
+                                  //~| NOTE missing `name` argument
 
 fn main() {
 }

